aws/table_aws_route53_traffic_policy.go

```diff
@@ -0,0 +1,261 @@
+		},
+		Columns: awsColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name that you specified when traffic policy was created.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The ID that Amazon Route 53 assigned to a traffic policy when it was created.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "version",
+				Description: "The version number that Amazon Route 53 assigns to a traffic policy.",
+				Type:        proto.ColumnType_INT,
+				Hydrate:     extractTrafficPolicyVersion,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "comment",
+				Description: "The comment that you specified when traffic policy was created.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getTrafficPolicy,
+			},
+			{
+				Name:        "document",
+				Description: "The definition of a traffic policy in JSON format.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getTrafficPolicy,
+			},
+			{
+				Name:        "type",
+				Description: "The DNS type of the resource record sets that Amazon Route 53 creates when you use a traffic policy to create a traffic policy instance.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getRoute53TrafficPolicyTurbotAkas,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listTrafficPolicies(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listTrafficPolicies")
+
+	// Create session
+	svc, err := Route53Service(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	input := &route53.ListTrafficPoliciesInput{
+		MaxItems: aws.String("100"),
+	}
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < 100 {
+			if *limit < 1 {
+				input.MaxItems = aws.String("1")
+			} else {
+				input.MaxItems = aws.String(fmt.Sprint(*limit))
+			}
+		}
+	}
+
+	// List call
+	pagesLeft := true
+	for pagesLeft {
+		result, err := svc.ListTrafficPolicies(input)
+		if err != nil {
+			plugin.Logger(ctx).Error("listTrafficPolicies", "ListTrafficPolicies_error", err)
+			return nil, err
+		}
+		for _, policies := range result.TrafficPolicySummaries {
+			_, err := listTrafficPolicyVersions(ctx, d, svc, policies.Id)
+			if err != nil {
+				return nil, err
+			}
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
```

Comment:
Should we handle context cancel logic here?

Reply:
I think it should be done in both places. Context may get cancelled even while fetching the policy. In that case, we can stop the calls here.